app/security.go

```diff
@@ -0,0 +1,50 @@
+package app
+
+import (
+	"strings"
+
+	"github.com/tsuru/tsuru/app/bind"
+)
+
+const suppressedEnv = "****"
+
+var sensitiveEnvs = []string{
+	"password",
+	"secret",
+	"passwd",
+	"api_key",
+	"apikey",
+	"token",
+	"credential",
+	"pwd",
+	"endpoint",
+}
+
+func isSensitiveName(name string) bool {
+	lowerStr := strings.ToLower(name)
+
+	for _, env := range sensitiveEnvs {
+		if strings.Contains(lowerStr, env) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (a *App) SuppressSensitiveEnvs() {
+	newEnv := map[string]bind.EnvVar{}
+	for key, env := range a.Env {
+		if !env.Public || isSensitiveName(env.Name) {
```

Comment:
It's better to rely only on the state declared by the user. Deducing whether the env var is sensitive based on its name may cause misunderstanding.

```suggestion
		if !env.Public {
```